lib/BBTools/utils/BBMapRunner.py

```diff
@@ -196,24 +201,78 @@ def _process_app_params_to_cli(self, io_params, app_params, output_dir, run_log,
                 unmapped_reads_out_file = os.path.join(output_dir, str(assembly_i)+'-'+io_params['out_obj_name']+'-'+'UNMAPPED'+'.FASTQ')
                 options.append('outu={}'.format(unmapped_reads_out_file))
 
-        # hard-code max threads for now
-        options.append('t={}'.format('4'))
+            # add user-config params
+            if app_params.get('input_parameter_suite'):
+                options.append(app_params['input_parameter_suite'])
+
+            if app_params.get('kmer_len') and int(app_params['kmer_len']) != 0:
```

Comment:
if `kmer_len` isn't an int, this'll throw a TypeError. Which is fine, but the app will look like it failed kinda mysteriously. Also probably fine as long as the spec enforces that it's an int that gets passed to it. Still might be worth capturing, or noting in case weird behavior happens.

Reply:
kmer_len gets a type check as int with proper error messaging in build_options() (and also is spec'ed as int in spec.json)

Reply:
👍